Kernel/Storage/IDEChannel.cpp

```diff
@@ -269,28 +302,37 @@ UNMAP_AFTER_INIT void IDEChannel::detect_disks()
     // There are only two possible disks connected to a channel
     for (auto i = 0; i < 2; i++) {
         // We need to select the drive and then we wait 20 microseconds... and it doesn't hurt anything so let's just do it.
+        IO::delay(20);
         m_io_group.io_base().offset(ATA_REG_HDDEVSEL).out<u8>(0xA0 | (i << 4)); // First, we need to select the drive itself
         IO::delay(20);
 
+        auto status = m_io_group.control_base().in<u8>();
+        if (status == 0x0) {
+            dbgln_if(PATA_DEBUG, "IDEChannel: No {} {} disk detected!", channel_type_string().to_lowercase(), channel_string(i));
+            continue;
+        }
+
         m_io_group.io_base().offset(ATA_REG_SECCOUNT0).out<u8>(0);
         m_io_group.io_base().offset(ATA_REG_LBA0).out<u8>(0);
         m_io_group.io_base().offset(ATA_REG_LBA1).out<u8>(0);
         m_io_group.io_base().offset(ATA_REG_LBA2).out<u8>(0);
         m_io_group.io_base().offset(ATA_REG_COMMAND).out<u8>(ATA_CMD_IDENTIFY); // Send the ATA_IDENTIFY command
 
-        // Wait for the BSY flag to be reset
-        while (m_io_group.control_base().in<u8>() & ATA_SR_BSY)
-            ;
-
-        if (m_io_group.control_base().in<u8>() == 0x00) {
-            dbgln_if(PATA_DEBUG, "IDEChannel: No {} {} disk detected!", channel_type_string().to_lowercase(), channel_string(i));
+        // Wait 10 second for the BSY flag to clear
```

Comment:
10 seconds seems like a hilariously long time to wait for a response to `ATA_IDENTIFY`

Reply:
Let's make it 2 seconds then, I think it's enough time for the drive to give a correct response.